src/axom/core/utilities/AnnotationMacros.hpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_ANNOTATION_MACROS_HPP_
+#define AXOM_ANNOTATION_MACROS_HPP_
+
+#include "axom/config.hpp"
+
+#ifndef AXOM_USE_CALIPER
+#include "axom/core/utilities/nvtx/interface.hpp"
+#endif
+
+/*!
+ * \def AXOM_PERF_MARK_FUNCTION( name )
+ * 
+ * \brief The AXOM_PERF_MARK_FUNCTION is used to annotate a function.
+ * 
+ * \param [in] name a user-supplied name to annotate the function.
+ * 
+ * \note Typically, the AXOM_PERF_MARK_FUNCTION is placed in the beginning of
+ *  the function to annotate.
+ * 
+ * 
+ * \warning The AXOM_PERF_MARK_FUNCTION can only be called once within a given
+ *  (function) scope.
+ * 
+ * 
+ * Usage Example:
+ * \code
+ *   void foo( )
+ *   {
+ *     AXOM_PERF_MARK_FUNCTION( "foo" );
+ *     ... 
+ *   }
+ * \endcode
+ */
+#if defined(AXOM_ENABLE_ANNOTATIONS) && defined(AXOM_USE_CALIPER)
```

Comment:
Is ``AXOM_USE_CALIPER`` defined anywhere in axom?

Should it be added to ``config.hpp.in`` (and obviously not defined yet)?

Reply:
No it is not defined. We can add it in the PR that adds Caliper.